DOMAIN.md

```diff
@@ -0,0 +1,134 @@
+## Introduction
+
+This document describes the domain model for service performance using
+an ML like language (actually F#) to describe the types that are found
+within the application, and their relationship to each other. Where
+necessary the validation of the types is documented.
+
+The terminology is mostly consistent with the app's models, but those
+models should be refactored to more closely follow this domain model.
+
+## Service Performance
+
+Service Performance provides a tool that allows organisations to publish
+metrics about their service transactions, and end users to view and track
+trends in the data.  The metrics collected revolve around the tracking of
+transactions in services, how many they receive, how they're received and
+how many of them are processed within a given month.
+
+Each month an organisation is expected to provide metrics for the previous
+month.  Some organisations do not collect some metrics, for instance when
+they don't have a call centre, they don't provide 'Calls received' metrics.
+These are said to be 'Not Provided'.
+
+For any given metric, the range of valid values are shown below.  There is
+either a value, the data is not provided, or it is not applicable for this
+service.
+
+```fsharp
+type NotApplicable = unit
+type NotProvided   = unit
+
+type MetricValue =
+  | Value of uint
+  | NotApplicable
+  | NotProvided
+```
+
+## Metrics
+
+### Transactions Received
+
+These metrics record the total number of transactions received by a service in a given month, and then a breakdown of the channels on which the transactions were received.  Not all services will receive transactions on all channels, and the other channel allows the capture of channels not currently supported.  There have been discussions about supporting multiple 'other' channels and this would likely require quite a deep restructure of the data model.
+
+```fsharp
+type TransactionsReceived = {
+  Total: MetricValue,
+  Phone: MetricValue,
+  Online: MetricValue,
+  Paper: MetricValue,
+  FaceToFace: MetricValue,
+  Other: MetricValue
+
+(*
+  Unlike the other group types, we never expect Transaction received
+  metrics to be non-applicable because it is the core metric we collect
+  and the other don't make any sense without it.
+*)
+type TransactionsReceivedGroup = TransactionsReceived
+```
+
+### Transactions Processed
+
+All services should report the total number of transactions that they have processed during a given month. As well as providing the total, they should also record the number of transactions that resulted in a positive outcome for the user - that is, it met their needs.  There is still some confusion in this area specifically around the semantics of "what does processed mean?" and "what is the users' intended outcome?".
```

Comment:
Is there a write up of user research around this that we could link to?

Reply:
I think this bit is all in flux.  If you ask KT she might be able to explain the new thinking about using different words to represent this.